tgext/utils/storage.py

```diff
@@ -0,0 +1,30 @@
+import uuid
+from os import path, mkdir
+
+from tg import config
+from tg.util.files import safe_filename
+
+
+def store(data):
+    folder_name = generate_uuid()
+    original_filename = safe_filename(data.filename)
+    destination = _specify_path(folder_name, original_filename)
+
+    with open(destination, mode='wb') as dest:
+        content = data.file.read()
```

Comment:
This should probably read and write the file in chunks (read 4k, write 4k until there is no more to read), to ensure that in case of a very big file we won't need to read it all in memory to save it.